src/core/imageGallery/ImageGallery.tsx

```diff
@@ -0,0 +1,264 @@
+  const [imageIndex, setImageIndex] = useState<number>(0);
+  const [selectedImageIndex, setSelectedImageIndex] = useState<number>(-1);
+  const [initialFocus, setInitialFocus] = useState<boolean>(false);
+  const [isLightboxVisible, setIsLightboxVisible] = useState<boolean>(false);
+
+  const {
+    copy: { closeButtonLabelText, next, previous },
+  } = useConfig();
+
+  const handleEnterKeyPress = (event) => {
+    if (withLightbox && event.key === 'Enter') {
+      setIsLightboxVisible(true);
+    }
+  };
+
+  const handleEscapeKeyPress = (event) => {
+    if (event.key === 'Escape') {
+      setIsLightboxVisible(false);
+    }
+  };
+
+  const handleImageCardFocus = (index: number) => {
+    setImageIndex(index);
+    // selected card index before opening the lightbox
+    setSelectedImageIndex(index);
+    barrierRef.current?.focus();
+    setInitialFocus(true);
+  };
+
+  useEffect(() => {
+    const lightbox = lightboxRef.current;
+    const focusableElements = lightbox?.querySelectorAll(
+      'button, [tabindex]:not([tabindex="-1"])',
+    );
+    const firstElement = focusableElements ? focusableElements[0] : null;
+    const lastElement = focusableElements
+      ? focusableElements[focusableElements.length - 1]
+      : null;
+
+    const handleTabKeyPress = (event) => {
+      if (focusableElements && event.key === 'Tab') {
+        if (event.shiftKey && document.activeElement === firstElement) {
+          event.preventDefault();
+          lastElement.focus();
+        } else if (!event.shiftKey && document.activeElement === lastElement) {
+          event.preventDefault();
+          focusableElements[0].focus();
+        }
+      }
+    };
+
+    if (isLightboxVisible) {
+      if (!initialFocus) {
+        barrierRef.current?.focus();
+        setInitialFocus(true);
+        lightbox?.addEventListener('keydown', handleTabKeyPress);
+        lightbox?.addEventListener('keydown', handleEscapeKeyPress);
+      }
+    } else {
+      setInitialFocus(false);
+      lightbox?.removeEventListener('keydown', handleTabKeyPress);
+      lightbox?.removeEventListener('keydown', handleEscapeKeyPress);
+    }
+  }, [
+    isLightboxVisible,
+    setIsLightboxVisible,
+    initialFocus,
+    setInitialFocus,
+    handleEscapeKeyPress,
+  ]);
+
+  useEffect(() => {
+    if (!isLightboxVisible && selectedImageIndex !== -1) {
+      const gridContainer = gridContainerRef.current;
+      const selectedCard = gridContainer?.querySelector(
+        `[id="${lightboxUid}-card-${selectedImageIndex}"]`,
+      );
+      if (selectedCard) {
+        selectedCard?.focus();
+      }
+    }
+  }, [isLightboxVisible, selectedImageIndex]);
+
+  const toggleLightbox = () => {
+    setIsLightboxVisible((prev) => !prev);
+  };
+
+  const handleImageCardClick = (
+    e: React.MouseEvent<HTMLDivElement, MouseEvent>,
+  ) => {
+    e.preventDefault();
+    if (withLightbox) {
+      toggleLightbox();
+    }
+  };
+
+  const getLightbox = useCallback(() => {
```

Comment:
This content should definitely be in a new component (or function).

Reply:
fixed